editor/dependency_editor.cpp

```diff
@@ -286,10 +286,23 @@ void DependencyEditorOwners::_list_rmb_clicked(int p_item, const Vector2 &p_pos,
 	file_options->clear();
 	file_options->reset_size();
 	if (p_item >= 0) {
-		if (owners->get_selected_items().size() == 1) {
-			file_options->add_icon_item(get_theme_icon(SNAME("Load"), SNAME("EditorIcons")), TTR("Open Scene"), FILE_OPEN);
+		PackedInt32Array selected_items = owners->get_selected_items();
+		bool only_scenes_selected = true;
+
+		for (int i = 0; i < selected_items.size(); i++) {
+			int item_idx = selected_items[i];
+			if (ResourceLoader::get_resource_type(owners->get_item_text(item_idx)) != "PackedScene") {
+				only_scenes_selected = false;
+				break;
+			}
+		}
+
+		if (only_scenes_selected) {
+			file_options->add_icon_item(get_theme_icon(SNAME("Load"), SNAME("EditorIcons")), TTRN("Open Scene", "Open Scenes", selected_items.size()), FILE_OPEN);
+		} else if (selected_items.size() == 1) {
+			file_options->add_icon_item(get_theme_icon(SNAME("Load"), SNAME("EditorIcons")), TTR("Open"), FILE_OPEN);
```

Comment:
Both the old and the new code is rather misleading, as both options are identical. Only the text is different. I think we could simplify this. And we could also rework the awkward return below in the process with an early return.

Reply:
Not sure I follow... You can open multiple scenes, but only one single resource.

Reply:
Both these options trigger `FILE_OPEN` with no additional information passed from this bit, only the text is different (even the icon is the same).

Reply:
Well, in fact, this is quite a lot of checking just to display a slightly different text. Perhaps we should just call the option "Open Selected"? What's the significance of writing "Open Scenes"?